File: practise/daliy practise/Scientific_Computing_With_Machine/OpenMP/Helloworld.f90
program main

  use omp_lib
  implicit none

  !integer thread_id, thread_count
  !integer OMP_GET_THREAD_NUM
  !OMP_GET_THREAD_NUM = 10
  !$omp parallel
  !private(thread_id, thread_count)
  !integer thread_id, thread_count
  !thread_id = omp_get_thread_num()
  !thread_count = omp_get_num_threads()
  !write (*,*) "Hello from thread number:", thread_id, "out of:" , thread_count
  print *, "Hello I am a thread"
  !$omp end parallel
end program main

File: practise/daliy practise/Scientific_Computing_With_Machine/OpenMP/helloworld.c
#include<stdio.h>
#include<omp.h>

int main(int argc, char* argv[]){
    int i = 1;
#pragma omp parallel
    {
        //i++;
        //printf("global value i: %d\n", i);
        int thread_id = omp_get_thread_num();
        int thread_cout = omp_get_num_threads();
        printf("Hello from thread number: %d out of: %d, and i: %d\n", thread_id, thread_cout, i);
        //printf("global value i: %d\n", i);
        //i++;
    }
    return 0;
}